### PR TITLE
User documentation for the new decompiler

### PR DESCRIPTION
**Pull request by NirmalManoj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
This is the documentation for the new context menu for the decompiler implemented by PR #2391 . 

**Test plan (required)**
1. Read all the documentation. Think if anything can be improved.
2. Make sure I didn't miss out anything.

### CODE REVIEW

**Comment by ITAYC0HEN** [comment]
Didn't review yet.

But I expect a real user documentation and not only menu reference. Means, a dedicated "r2book"-like page for the decompiler widget. Yes, the decompilers are different from each other and we only provide the interface, and still.

See for example Ghidra's. Partial screenshot:
image

Full source:
https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Decompiler/src/main/help/help/topics/DecompilePlugin/Decompiler.htm


and there are few nice things in IDA's
https://www.hex-rays.com/products/decompiler/compare/compare_vs_disassembly/
https://www.hex-rays.com/products/decompiler/manual/index.shtml



I also expect to refer the readers to available decompilers -> https://github.com/radareorg/cutter-plugins#decompilers

Also to mention that r2ghidra is the official plugin maintained but the devs of cutter and r2.
I also would like to see some tips for how to create a new decompiler. I'd like to see the information for devs about the structure of the annotations (maybe better to be put in a mutual place with r2).

Also to show how it can work with debugger and list common issue s(oh boy there are...), how it can be synced with the disassembly, and more and more. I want the reader to go to a journey in a book to understand the decompiler in cutter.


As it is now, I think it is far from enough. It is simply a context menu reference...

**Review by karliss — CHANGES_REQUESTED** [review]

**Comment by NirmalManoj** [comment]
@ITAYC0HEN Where do you want such a page that introduces the Cutter's decompiler to appear?


> also would like to see some tips for how to create a new decompiler. I'd like to see the information for devs about the structure of the annotations

I can explain the structure of the annotations. But what did you actually mean by the second sentence?

**Comment by NirmalManoj** [comment]
> I know @karliss might have different opinions so let's hear it before you move forward for the dev documentation (user docs should proceed as discussed)

@karliss Do you have different opinions on the documentation for developers?

File: docs/source/user-docs/menus/decompiler-context-menu/debug-context-menu.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Debug Context Menu
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu/debug-context-menu.rst]: Added this as a section under the decompiler-context-menu page and removed these files.

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+   decompiler-context-menu/*
+
+Copy
+----------------------------------------
+**Description:** If text is selected, copy selected text. Otherwise, if a word is highlighted, copy that word. In all other cases, copy the line under the cursor.
+
+**Steps:**  Right-click on a selected text and choose ``Copy``  
+
+**Shortcut:** :kbd:`Ctrl` + :kbd:`C`
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu/debug-context-menu.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+==============================
+
+Continue Until Line
+----------------------------------------
+**Description:** Continue the execution of the program until it reaches the offset in the selected line. The program is not guaranteed to ever reach this address and will keep running until exited or until reached another breakpoint. If other breakpoints hit before reaching this line, they will be triggered and pause the execution. *This option is only available on Debug or Emulation modes*.      
+
+**Steps:** While in Debug or Emulation modes, right-click and address and choose ``Debug -> Continue until line``.
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu/debug-context-menu.rst]: Fixed

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+Show the selected item in another widget
+----------------------------------------
+**Description:** Show the selected item in another opened widget, or open a new one. Items include functions, global variables, and constant variables under the cursor.
+
+**Steps:**  Right-click on an item and choose the ``Show [item] in`` sub-menu.
+
+Add and Edit Comment
+----------------------------------------
+**Description:** Add a comment in the location under the cursor or edit the comment in the location under the cursor. Edit comment option is available only for user defined comments.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:41]:
```suggestion
**Description:** Add a comment in the location under the cursor or edit the comment in the location under the cursor. The ``Edit Comment`` option is only available for user defined comments.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:** Right-click and choose `Add Comment` or `Edit Comment`.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:43]: need to use double backtics

```suggestion
**Steps:** Right-click and choose ``Add Comment`` or ``Edit Comment``.
```


Check all other places for the same mistake

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+----------------------------------------
+**Description:** Delete the comment at the location under the cursor. This option is available only for user defined comments.
+
+**Steps:** Right-click on an instruction with a user-defined comment and choose ``Delete comment``  
+
+Rename function
+----------------------------------------
+**Description:** Rename the function under cursor. 
+
+**Steps:** Right-click the function name and choose ``Rename function <name of function>``
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:57]:
```suggestion
**Steps:** Right-click on a function name and choose ``Rename function <name of function>``  
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:** Right-click the function name and choose ``Rename function <name of function>``  
+
+**Shortcut:** :kbd:`N`
+
+Give a name or rename global variables
+----------------------------------------
+**Description:** Give a name or rename the global variable under the cursor.
+
+**Steps:** Right-click the global variable and choose ``Add name to <address of global variable>`` or ``Rename [name]``.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:65]: mixed `<placeholder>` and `[placeholder]`

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+**Shortcut:** :kbd:`N`
+
+Delete the name of global variable
+----------------------------------------
+**Description:** Delete the name of the global variable under the cursor.
+
+**Steps:** Right-click the global variable and choose ``Remove [name of global variable]``.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:73]: also, notice the mixture of `<placeholder>` and `[placeholder]` please also check it across the PR

```suggestion
**Steps:** Right-click on a global variable and choose ``Remove [name of global variable]``.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+Rename Function Variable
+----------------------------------------
+**Description:** Rename local variables and arguments in the decompiled function. Note that this option is available only for function variables defined in the disassembly.
+
+**Steps:** Right-click the variable and choose ``Rename variable [name]``.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:79]:
```suggestion
**Steps:** Right-click on a variable and choose ``Rename variable [name]``. 
```

this mistake is repeating, please check across the entire PR

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+----------------------------------------
+**Description:** Rename local variables and arguments in the decompiled function. Note that this option is available only for function variables defined in the disassembly.
+
+**Steps:** Right-click the variable and choose ``Rename variable [name]``. 
+
+**Shortcut:** :kbd:`N` 
+
+Edit Local Variables and Arguments
+----------------------------------------
+**Description:** Rename or set the types of the function's variables and arguments. Note that this option is available only for function variables defined in the disassembly.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:85]:
```suggestion
**Description:** Rename or set the types of the function's variables and arguments. Note that this option is available only for function variables and arguments defined in the disassembly.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:** Right-click on a reference and choose ``Show X-Refs``  
+
+**Shortcut:** :kbd:`X`
+
+Manage Breakpoints in the Context Menu
+=======================================
+Add or Remove Breakpoint in the line
+----------------------------------------
+**Description:** Add a breakpoint at the earliest offset in the line under the cursor. If you use the shortcut and if multiple breakpoints are present in a line, all of them will be removed.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:103]:
```suggestion
**Description:** Add a breakpoint at the earliest offset in the line under the cursor. If you use the keyboard shortcut to remove a breakpoint, all the breakpoints defined in the line will be removed.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:** Right-click on an instruction and choose ``Breakpoint -> [Add][Remove] [all] breakpoint[s in line]``  
+
+**Shortcut:** :kbd:`F2`  
+
+Advanced Breakpoint Dialog
+----------------------------------------
+**Description:** Open the advanced breakpoint dialog. This dialog lets you define not only a regular breakpoint in this address, but also a Hardware breakpoint, a conditional breakpoint, and more.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:111]: Either "**H**ardware breakpoint" and "**C**onditional breakpoint" or don't capitalize any

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:** Right-click on an instruction and choose ``Breakpoint -> Advanced breakpoint``. If multiple breakpoints are present in the line, you will be able choose the breakpoint you want to edit from the ``Edit breakpoint`` submenu.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:113]: see :menuselection: https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-menuselection

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+
+**Shortcut:** :kbd:`Ctrl-F2`
+
+Debug Context Menu
+=======================================
+Continue Until Line
+----------------------------------------
+**Description:** Continue the execution of the program until it reaches the offset in the selected line. The program is not guaranteed to ever reach this address and will keep running until exited or until reached another breakpoint. If other breakpoints hit before reaching this line, they will be triggered and pause the execution. *This option is only available on Debug or Emulation modes*.      
+
+**Steps:** While in Debug or Emulation modes, right-click and address and choose ``Debug -> Continue until line``.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:123]:
```suggestion
**Steps:** While in Debug or Emulation modes, right-click on an address and choose ``Debug -> Continue until line``.  
```

Comment (ITAYC0HEN) [reply docs/source/user-docs/menus/decompiler-context-menu.rst:123]: see :menuselection:

**Review by ITAYC0HEN — CHANGES_REQUESTED** [review]
Very nice Nirmal! :)

I will have more review comments, especially about `features.rst`

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+=======================================
+Continue Until Line
+----------------------------------------
+**Description:** Continue the execution of the program until it reaches the offset in the selected line. The program is not guaranteed to ever reach this address and will keep running until exited or until reached another breakpoint. If other breakpoints hit before reaching this line, they will be triggered and pause the execution. *This option is only available on Debug or Emulation modes*.      
+
+**Steps:** While in Debug or Emulation modes, right-click and address and choose ``Debug -> Continue until line``.  
+
+Set Program Counter (PC)
+----------------------------------------
+**Description:** Set the Program Counter of the debuger to the current offset. For example, on an Intel 64bit program, Cutter will set the value of the RIP register to the current address.  *This option is only available on Debug or Emulation modes*.
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:127]:
```suggestion
**Description:** Set the Program Counter of the debugger to the current offset. For example, on an Intel 64bit program, Cutter will set the value of the RIP register to the current address.  *This option is only available on Debug or Emulation modes*.  
```

Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 14 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 13 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 12 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Made it lower case.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 11 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 10 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 9 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Fixed in multiple places in the document. Please check again.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 8 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Checked across PR.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 6 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 5 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 4 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+
+Copy Address of Reference
+----------------------------------------
+**Description:** Copy the address of the reference under cursor. References include functions, global variables, and constant variables with an address.
+
+**Steps:**  Right-click on a location and choose ``Copy  address of [<name of reference>](<address>)``  
+
+**Shortcut:** :kbd:`Ctrl-Shift-C`
+
+Show the code in another widget
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst:27]: I could find this only one location iirc. Please check again.

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+----------------------------------------
+**Description:** Show the code under cursor in another opened widget, or open a new one. If a non-decompiler widget is chosen, the address mapped to the portion of code under the cursor will be opened in that widget.
+
+**Steps:**  Right-click on an item and choose the ``Show in`` sub-menu.
+
+Show the selected item in another widget
+----------------------------------------
+**Description:** Show the selected item in another opened widget, or open a new one. Items include functions, global variables, and constant variables under the cursor.
+
+**Steps:**  Right-click on an item and choose the ``Show [item] in`` sub-menu.
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Fixed.

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:**  Right-click on a location and choose ``Copy  address of [<name of reference>](<address>)``  
+
+**Shortcut:** :kbd:`Ctrl-Shift-C`
+
+Show the code in another widget
+----------------------------------------
+**Description:** Show the code under cursor in another opened widget, or open a new one. If a non-decompiler widget is chosen, the address mapped to the portion of code under the cursor will be opened in that widget.
+
+**Steps:**  Right-click on an item and choose the ``Show in`` sub-menu.
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Rephrased to ```Right-click on an item and go to the :menuselection:`Show in` submenu. You can choose a widget or open a new widget from here.```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+----------------------------------------
+**Description:** Copy the instruction address mapped to the part of code under the cursor.
+
+**Steps:**  Right-click on a location and choose ``Copy instruction address(<address>)``
+
+Copy Address of Reference
+----------------------------------------
+**Description:** Copy the address of the reference under cursor. References include functions, global variables, and constant variables with an address.
+
+**Steps:**  Right-click on a location and choose ``Copy  address of [<name of reference>](<address>)``
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed. Please check now.

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 20]
```diff
@@ -0,0 +1,129 @@
+Decompiler Context Menu 
+==============================
+The decompiler context menu contains actions that are available for the location under cursor.
+
+Copy
+----------------------------------------
+**Description:** If text is selected, copy selected text. Otherwise, if a word is highlighted, copy that word. In all other cases, copy the line under the cursor.
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:**  Right-click on a selected text and choose ``Copy``  
+
+**Shortcut:** :kbd:`Ctrl-C`  
+
+Copy Instruction Address
+----------------------------------------
+**Description:** Copy the instruction address mapped to the part of code under the cursor.
+
+**Steps:**  Right-click on a location and choose ``Copy instruction address(<address>)``
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Rephrased to ```Copy the address of the instruction mapped to the part of code under the cursor.```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+----------------------------------------
+**Description:** If text is selected, copy the selected text to the clipboard. If a word is highlighted, copy that word. In all other cases, copy the line under the cursor.
+
+**Steps:**  Right-click on a selected text and choose ``Copy``
+
+**Shortcut:** :kbd:`Ctrl-C`  
+
+Copy Instruction Address
+----------------------------------------
+**Description:** Copy the address of the instruction mapped to the part of code under the cursor.
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:15]:
```suggestion
**Description:** Copy the address of the instruction mapped to the part of the code under the cursor.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:**  Right-click on the portion of code for which you want the instruction's address and choose ``Copy instruction address (<address>)``
+
+Copy Address of Reference
+----------------------------------------
+**Description:** Copy the address of the reference under cursor. References include functions, global variables, and constant variables with an address.
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:21]:
```suggestion
**Description:** Copy the address of the reference under the cursor. References include functions, global variables, and constant variables with an address.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+
+Show the selected item in another widget
+----------------------------------------
+**Description:** Show the selected item in another opened widget, or open a new one. Items include functions, global variables, and constant variables under the cursor.
+
+**Steps:**  Right-click on an item and go to the submenu :menuselection:`Show <item> in` or :menuselection:`<function name> (<address>)`. You can choose a widget or open a new widget from here.
+
+Add and Edit Comment
+----------------------------------------
+**Description:** Add a comment in the location under the cursor or edit the comment in the location under the cursor. The ``Edit comment`` option is only available for user defined comments.
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:41]:
```suggestion
**Description:** Add a comment in the location under the cursor or edit the comment in the location under the cursor. The ``Edit comment`` option is only available for user-defined comments.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+
+**Steps:** Right-click and choose ``Add Comment`` or ``Edit Comment``.
+
+**Shortcut:** :kbd:`;`
+
+Delete a Comment
+----------------------------------------
+**Description:** Delete the comment at the location under the cursor. This option is only available for user defined comments.
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:49]:
```suggestion
**Description:** Delete the comment at the location under the cursor. This option is only available for user-defined comments.
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 26]
```diff
@@ -0,0 +1,129 @@
...
+----------------------------------------
+**Description:** Rename or set the types of the function's variables and arguments. Note that this option is available only for function variables and arguments defined in the disassembly.
+
+**Steps:** Right-click on a variable and choose ``Edit variable <name>``.
+
+**Shortcut:** :kbd:`Y`
+
+Show Cross References
+----------------------------------------
+**Description:** Show X-Refs to the reference under cursor. This option will open Cutter's X-Refs dialog in which you will be able to see a list of X-Refs from and to the address of the reference. You can also see a preview of each cross-reference to quickly inspect the different usages. Note that references refer to function names, global variables, and constant variables with an address.
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:93]:
```suggestion
**Description:** Show X-Refs to the reference under the cursor. This option will open Cutter's X-Refs dialog in which you will be able to see a list of X-Refs from and to the address of the reference. You can also see a preview of each cross-reference to quickly inspect the different usages. Note that references refer to function names, global variables, and constant variables with an address.  
```

File: docs/source/user-docs/menus/decompiler-context-menu.rst [hunk 27]
```diff
@@ -0,0 +1,129 @@
...
+
+Give a name or rename global variables
+----------------------------------------
+**Description:** Give a name or rename the global variable under the cursor.
+
+**Steps:** Right-click the global variable and choose ``Add name to <address of global variable>`` or ``Rename <name>``.
+
+**Shortcut:** :kbd:`N`
+
+Delete the name of global variable
```
Comment (Surendrajat) [inline docs/source/user-docs/menus/decompiler-context-menu.rst:69]:
```suggestion
Delete the name of a global variable
```

**Review by Surendrajat — CHANGES_REQUESTED** [review]
Hail Grammarly :)

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 27 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested.

File: docs/source/user-docs/menus/decompiler-context-menu.rst (hunk 26 above)
Comment (NirmalManoj) [reply docs/source/user-docs/menus/decompiler-context-menu.rst]: Changed as suggested. This and every other suggestion you have made.

**Comment by karliss** [comment]
> @karliss Do you have different opinions on the documentation for developers?

I have no objections against having a page in developer docs describing high level structure of decompiler in cutter and R2. What are the components involved, what are their main responsibilities and how they communicate with each other.

It shouldn't duplicate the API documentation and try to describe each method or each annotation. It doesn't need to be a tutorial for people who can barely program.

To summarize here is what I would like to see for developer docs:
* Document the Decompiler interface using doxygen
* Page in developer docs describing high level structure of decompiler in Cutter
   * What are the components involved (r2 plugin, r2, Cutter, Cutter plugin[cutter plugin impl, Cutter Decompiler impl])
   * Role of each component. Described using human language instead of literal translation code into English.  Description like "This module calls this function, then it calls that function using the result function and then it calls one more function" is in my opinion worse than no documentation and reading the code directly. At least when reading the code it's up to date.
   * How the components communicate with each other, at high level, details should be in API docs
   * Checklist of what needs to be implemented (Cutter Plugin,Cutter Decompiler, r2 plugin optional but recommended)

**Comment by NirmalManoj** [comment]
@ITAYC0HEN I am not sure if I could make developer documentation in the next two days like Karliss wants. So I'm thinking of doing it after my project in another PR. User documentation is important, so I don't want developer doc to block this PR from getting merged.

**Comment by ITAYC0HEN** [comment]
@NirmalManoj this is fine :) I prefer you relaxed and focused on the things you must do, rather then submitting ahlf-baked dev docs. So, no problems - let's keep dev docs for a following PR :)

**Review by ITAYC0HEN — APPROVED** [review]

File: docs/source/user-docs/features.rst [hunk 28]
```diff
@@ -1,18 +1,19 @@
 Features
 =============
-You can read about the exciting features in Cutter here.
+This section in the user docs gives a more detailed information about different features in Cutter. These are the different
+utilities and vies that be used and controlled by you during your session. As more familiar and comfort you feel with the
```
Comment (NirmalManoj) [inline docs/source/user-docs/features.rst:4]: *views

Comment (NirmalManoj) [reply docs/source/user-docs/features.rst:4]: utilities and _views_ that _can_ be used and

Comment (ITAYC0HEN) [reply docs/source/user-docs/features.rst:4]:
```suggestion
utilities and views that be used and controlled by you during your session. As more familiar and comfort you feel with the
```

Comment (NirmalManoj) [reply docs/source/user-docs/features.rst]: Shouldn't it be _that **can** be_?

File: docs/source/user-docs/features.rst [hunk 29]
```diff
@@ -0,0 +1,19 @@
+Features
+=============
+This section in the user docs gives a more detailed information about different features in Cutter. These are the different
+utilities and views that be used and controlled by you during your session. As more familiar and comfort you feel with the
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/features.rst:4]:
```suggestion
utilities and views that can be used and controlled by you during your session. As more familiar and comfort you feel with the
```

**Review by karliss — APPROVED** [review]

**Review by Surendrajat — APPROVED** [review]

**Pull request merged**